version: '3.8'

services:
  db:
    image: postgres:14
    container_name: quiz_postgres
    environment:
      POSTGRES_DB: quiz_db
      POSTGRES_USER: quiz_user
      POSTGRES_PASSWORD: quiz_password
    volumes:
      - db_data:/var/lib/postgresql/data
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - "64322:5432"  # Проброс внешнего порта на внутренний порт PostgreSQL
    networks:
      - quiz_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U quiz_user -d quiz_db"]
      interval: 10s
      timeout: 5s
      retries: 5

  web:
    build:
      context: ./app
      dockerfile: Dockerfile
    container_name: quiz_flask
    environment:
      FLASK_ENV: development
      DATABASE_URL: postgres://quiz_user:quiz_password@db:5432/quiz_db
    ports:
      - "5000:5000"
    depends_on:
      db:
        condition: service_healthy
    networks:
      - quiz_network

volumes:
  db_data:

networks:
  quiz_network:
    driver: bridge
